rpcs3/util/vm_native.cpp

```diff
@@ -218,8 +218,14 @@ namespace utils
 		int _prot = PROT_NONE;
 		switch (prot)
 		{
+#if defined(ARCH_ARM64) && defined(__APPLE__)
+		// Hack: define all r and rw pages as rw on 16k page platform to prevent page boundary issues
```

Comment:
Why not adjust address and size instead?

Reply:
It will cause adjacent pages' permissions to be overwritten. If I set the first 4kb to R only then it will set the value for the whole 16k page. If the second 4kb is supposed to be RW then it will break.

This problem existed in the PPU port as well, the workaround there was to explicitly set RW on mmap calls too.

Reply:
Well, why not set the R for the whole 16k page?

Reply:
Wouldn't that break code that's expecting to be able to write to the 4kb-16kb segment?

Reply:
If this call is coming from RSX then not.

Reply:
I was segfaulting in SPU generated code, I guess this is not needed for RSX.

I had tried just changing the page size in `vm` before and it breaks a lot of stuff, seems like a big job to fix.

Reply:
RSX relies on memory protection, it can't work correctly without it!